pytorch_lightning/metrics/classification.py

```diff
@@ -230,7 +230,7 @@ def __init__(
             >>> target = torch.tensor([0, 1, 2, 2])
             >>> metric = Precision()
             >>> metric(pred, target)
-            tensor(1.)
+            tensor(0.75)
```

Comment:
was this fixed in code? because 1. was the real return value...

Reply:
I realized now that the example is bad, since `pred` contains more classes than `target`, which seems to have unexpected behavior if the user does not set `num_classes` at construction. Either a new example or this we change `metric = Precision()` to `metric=Precision(num_classes=4)`.

Reply:
shouldn't we take the max of both and use that as the num_classes?

Reply:
So sklearn gives a warning if the user does this: `UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use zero_division parameter to control this behavior.`
I guess there a reason why they do not take max of both.

Reply:
like
```
if not num_classes:
    num_classes = max(max(pred), max(target)) + 1
```

Reply:
I have update code now, to account for this.
For safety: @justusschock is there a reason for this behavior that you are aware of?